cmd/juju/model/destroy.go

```diff
@@ -216,6 +225,26 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
+	// Check if the model has an sla auth.
+	if !slaIsSet {
+		return nil
+	}
+
+	bakeryClient, err := c.BakeryClient()
```

Comment:
Consider moving lines 233-246 to an unexported method, `func (c *destroyCommand) deleteBudgetAllocation(modelDetails) error`. `deleteBudgetAllocation` can then return early on error, which would be logged here with a warning.